05/app.js

```diff
@@ -0,0 +1,60 @@
+const formEl = document.querySelector('form');
+const errorsBox = formEl.querySelector('.messages');
+formEl.noValidate = true;
+
+formEl && formEl.addEventListener('submit', submitForm);
+
+function submitForm(e) {
+  e.preventDefault();
+  const errors = [];
+  errors.length = 0;
+  errorsBox.innerHTML = '';
+  const inputsList = [...formEl.elements];
+
+  inputsList.forEach((input) => {
+    validateForm(input, errors);
+  });
+
+  if (errors.length > 0) {
+    showErrors(errors);
+  } else {
+    alert('Formularz został wysłany!');
+    inputsList.forEach((input) => {
+      if (input.type !== 'submit') {
+        input.value = '';
+      }
+    });
+  }
+}
+
+function validateForm(input, errors) {
```

Comment:
Spróbuj napisać funkcje do walidacji, która będzie bardziej elastyczna - tak, aby można było jej użyć w innym zadaniu/projekcie. Taka funkcja powinna przyjmować zasady dot. walidacji oraz wartości do sprawdzenia, a zwracać błędy. Wtedy będzie to wystarczająco uniwersalne.

Możesz się wzorować tym artykułem:
https://devmentor.pl/b/walidacja-formularza-w-javascript

PS. Traktuj to jako zadanie dodatkowe ;)